pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -2870,3 +2877,47 @@ func releaseAppNetworkResources(ctx *zedrouterContext, key string,
 	}
 	publishAppNetworkStatus(ctx, status)
 }
+
+func isDNSServerChanged(new *types.DeviceNetworkStatus) bool {
+	for _, port := range new.Ports {
+		if port.IsMgmt {
+			if _, ok := dnsServers[port.Name]; !ok {
+				if len(port.DnsServers) > 0 { // just assigned
+					dnsServers[port.Name] = port.DnsServers
+				}
+			} else {
+				if len(port.DnsServers) != 0 {
+					// new one has different entries, and not the Internet disconnect case
+					if len(dnsServers[port.Name]) != len(port.DnsServers) {
+						return true
+					}
+					for idx, server := range port.DnsServers { // compare each one and update if changed
+						if server.Equal(dnsServers[port.Name][idx]) == false {
+							log.Infof("isDnsServerChanged: intf %s exist %v, new %v\n",
+								port.Name, dnsServers[port.Name], port.DnsServers)
+							dnsServers[port.Name] = port.DnsServers
+							return true
+						}
+					}
+
+				}
+			}
+		}
+	}
+	return false
+}
+
+func doDnsmasqRestart(ctx *zedrouterContext) {
```

Comment:
It would be more efficient if we determine which ports changed their DNS and only update/restart fir the network instances which use those ports. However, I just re-realize we don't separate the DNS per uplink. Need to file a separate bug on that.

Reply:
ok.